### PR TITLE
PropertyGraph accessors for indexes

### PR DESCRIPTION
**Pull request by arekay**
* Add property index accessors.
* Include support for `uint8`.

### CODE REVIEW

File: libgalois/include/katana/PropertyGraph.h [hunk 1]
```diff
@@ -762,6 +762,27 @@ class KATANA_EXPORT PropertyGraph {
   edge_indexes() const {
     return edge_indexes_;
   }
+
+  // Returns true of an index exists for the named property
```
Comment (micah-white) [inline libgalois/include/katana/PropertyGraph.h:766]: *if
obviously don't need to recommit for that lol

File: libgalois/include/katana/PropertyIndex.h [hunk 2]
```diff
@@ -30,13 +30,14 @@ class KATANA_EXPORT PropertyIndex {
...
   using set_key_type = typename std::variant<
-      IndexID, bool, int64_t, double_t, std::string_view*>;
+      IndexID, bool, uint8_t, int64_t, double_t, std::string_view*>;
 
   // PropertyIndex::iterator returns a sequence of node or edge ids.
   using base_iterator = typename std::multiset<set_key_type>::iterator;
   class iterator : public base_iterator {
   public:
     explicit iterator(const base_iterator& base) : base_iterator(base) {}
+    iterator() : base_iterator(0) {}
```
Comment (amberhassaan) [inline libgalois/include/katana/PropertyIndex.h:40]: We're making assumptions about constructor for `multiset::iterator`, which is not part of public STL API and can change in the future. This is risky.

**Review by micah-white — APPROVED** [review]
LGTM

**Review by amberhassaan — APPROVED** [review]

File: libgalois/include/katana/PropertyGraph.h (hunk 1 above)
Comment (arekay) [reply libgalois/include/katana/PropertyGraph.h:766]: Yeah, noticed that today - will *try* to keep in mind to fix it.

File: libgalois/include/katana/PropertyIndex.h (hunk 2 above)
Comment (arekay) [reply libgalois/include/katana/PropertyIndex.h:40]: There is a `do_all` path which requires this constructor. I intend to spend more time on figuring out the right way, but agree that this is risky.

**Pull request merged**